packages/drivers/local-driver/src/localDocumentDeltaConnection.ts

```diff
@@ -162,12 +161,14 @@ export class LocalDocumentDeltaConnection
         });
 
         this.on("newListener", (event, listener) => {
-            this.socket.on(
-                event,
-                (...args: any[]) => {
-                    this.emitter.emit(event, ...args);
-                });
-            this.emitter.on(event, listener);
+            if (!this.subscribedEvents.has(event)) {
+                this.subscribedEvents.add(event);
```

Comment:
Checking `.listeners("...").length` or `.listenerCount("...")` is safer here, since we don't handle listener removal to update the set.

Reply:
There are no such methods on socket (IWebSocket) - it does not extends EventEmitter (at list on public surface).
Checking it on self is not safe because we do unsubscribe events from LocalDocumentDeltaConnection - that's how tests fail with removal of DeltaConnection